pkg/runtime/reconciler.go

```diff
@@ -268,11 +269,24 @@ func (r *resourceReconciler) Sync(
 			rlog.Info("updated resource")
 		}
 	}
-	if ackcompare.IsNotNil(latest) {
+	rlog.Enter("rm.LateInitialize")
+	lateInitializedLatest, lateInitializeRequeueErr := rm.LateInitialize(ctx, latest)
+	rlog.Exit("rm.LateInitialize", lateInitializeRequeueErr)
+	if ackcompare.IsNotNil(lateInitializedLatest) {
 		err = r.patchResource(ctx, desired, latest)
 		if err != nil {
 			return err
 		}
+		if lateInitializeRequeueErr != nil {
+			var requeueCause string
+			if lateInitializeRequeueErr.Error() == "" {
```

Comment:
hmrph... this doesn't seem right... 

Are you trying to determine if the error returned from `LateInitialize` is a requeue? If so, there's a better way to do that using the standard `errors` package's `As` function:

```go
var requeueNeededAfter *requeue.NeededAfter
if errors.As(lateInitializeErr, &requeueNeededAfter) {
...
}
```

However, in this case, the `resourceReconciler.handlerReconcileError` method already handles this for you:

https://github.com/aws-controllers-k8s/runtime/blob/055b089c6a508317ad4bb57ce037dc55061e1829/pkg/runtime/reconciler.go#L522-L554

So all you need to do here is:

```go
if lateInitializeRequeueErr != nil {
    return lateInitializeRequeueErr
}
```

And because `handleReconcileError` is handling all the logging and requeueing, etc, you can probably simplify this whole code block to just this:

```go
rlog.Enter("rm.LateInitialize")
latest, err = rm.LateInitialize(ctx, desired, latest)
rlog.Exit("rm.LateInitialize", err)
if err != nil {
    if ackcompare.IsNotNil(latest) {
        _ = r.patchResource(ctx, desired, latest)
    }
    return err
}

if ackcompare.IsNotNil(latest) {
    _ = r.patchResource(ctx, desired, latest)
    // Do the conditions stuff...
}
```

Reply:
Thanks, the code is much more simple now.